Fixed editing of testfester

sarastray opened the pull request:
- added update of oppgaver in controller/oppgaver.js 
addTestfester.jsx:
-removed handleChange and handleSaveOppgaver
-common handleSaveAll function for all changes
- fixed problem with date
- some UX design fixes

sarastray committed:
fixed editing of testfester

- added update of oppgaver in controller/oppgaver.js
addTestfester.jsx:
-removed handleChange and handleSaveOppgaver
-common handleSaveAll function for all changes
- fixed problem with date
- some UX design fixes

TriggeredBanana reviewed (changes requested):
Overall very good PR! Just some nitpick and security issues to fix before I merge it:

1. Please add validation to make sure a user is logged in before being able to edit a task in the "updateOppgaver" function, example: if (!bruker) { return res.status(401).json({ error: "Ikke innlogget" }); }
Also verify that the user is the owner of the testfest (or superuser) before they are able to edit the tasks. (At the moment, any user can edit any task)

2. Add a bit extra validation to check that there is a task, example: if (isNaN(oppgaveID) || oppgaveID <= 0)

3. You removed multiple validations from the code in "testfester.js". If this was done on purpose, please explain. Otherwise, please keep validation in place as it makes sure the program runs smoothly or alternatively gives a valid error if something goes wrong.
You also removed the login check if (!bruker) before the authorization check in "deleteTestfester" function.

4. You changed changed some validations such as "if (!bruker.ErSuperbruker)" to "if (bruker.ErSuperbruker !== 1)" which does the same, but in a somewhat worse way.

5. You changed  "if (bruker.BrukerID !== eierId && !bruker.ErSuperbruker)" to "if (Number(bruker.BrukerID) !== Number(eierId) && !erSuper)". The "Number" validation addon is good! But the "erSuper" is an incorrect variable. We use "ErSuperbruker".

6. Remove "console.log" parts, they just clutter up the web console, and might sometimes contain information that should not be shared. Great for testing, but not for deployment :)

sarastray committed:
fixed validation

-Fixed validation that got removed
-Added validation that was missing
-Removed console.log

TriggeredBanana committed:
Security improvements

- Improves validation and removes console log.
